src/MongoDB/Manager.c

```diff
@@ -279,7 +279,9 @@ PHP_METHOD(Manager, selectServer)
 	}
 
 	readPreference = phongo_read_preference_from_zval(zreadPreference TSRMLS_CC);
-	selected_server = mongoc_topology_select(intern->client->topology, MONGOC_SS_READ, readPreference, MONGOC_SS_DEFAULT_LOCAL_THRESHOLD_MS, &error);
+	/* TODO: Remove (mongoc_read_prefs_t *) casting once
+	 * https://github.com/mongodb/mongo-c-driver/pull/315 is merged. */
+	selected_server = mongoc_client_select_server(intern->client, true, (mongoc_read_prefs_t *) readPreference, &error);
```

Comment:
I've just checked the libmongoc docs as well, and it states:

```
      <tr><td><p>for_writes</p></td><td><p>Whether to choose a server suitable for writes or reads.</p></td></tr>
      <tr><td><p>prefs</p></td><td><p>An optional <code xref="mongoc_read_prefs_t">mongoc_read_prefs_t</code>. If <code>for_writes</code> is true, <code>prefs</code> must be NULL. Otherwi
se, use <code>prefs</code> to customize server selection, or pass NULL to use the read preference configured on the client.</p></td></tr>
```

As we don't know yet whether people want to read and/or write, I'm not sure what to do here. The original topology_select uses MONGOC_SS_READ, which I guess means a non-writeable server perhaps?

We could set the "true" depending on whether the readPreference is RP_PRIMARY, and only then? But then an RP of PRIMARY_PREFERED or SECONDARY_PREFERED, which ends up selecting a non-master server wouldn't allow for writes...